src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -863,16 +863,22 @@ static void
 EnsureLocalTableEmptyIfNecessary(Oid relationId, char distributionMethod,
 								 bool viaDepracatedAPI)
 {
-	if (viaDepracatedAPI)
-	{
-		EnsureLocalTableEmpty(relationId);
-	}
-	else if (distributionMethod != DISTRIBUTE_BY_HASH &&
-			 distributionMethod != DISTRIBUTE_BY_NONE)
-	{
-		EnsureLocalTableEmpty(relationId);
-	}
-	else if (!RegularTable(relationId))
+	/* we don't support copying local data via deprecated APIs */
+	bool shouldEnsureLocalTableEmpty = viaDepracatedAPI;
+
+	bool distributionMethodRequiresEmptyTable =
+		(distributionMethod != DISTRIBUTE_BY_HASH &&
+		 distributionMethod != DISTRIBUTE_BY_NONE);
+
+	/* we only support hash distributed tables and reference tables for initial data loading */
+	shouldEnsureLocalTableEmpty |= distributionMethodRequiresEmptyTable;
```

Comment:
this syntax is shorter, but I find it harder to digest than an if statement.

so maybe just:
```c
if (distributionMethodRequiresEmptyTable)
{
    shouldEnsureLocalTableEmpty = true;
}
```